rstudio/rstudio
# place python on the PATH for new terminal sessions

### Intent

Users can configure the version of Python to be used by RStudio and reticulate in a couple ways:

- By explicitly setting the `RETICULATE_PYTHON` environment variable;
- Via the UI, with global / project-specific options.

However, none of these configured versions of Python would propagate as the "default" to the terminal. This PR tries to make that happen.

Addresses https://github.com/rstudio/rstudio/issues/9188.

### Approach

Making this work was a bit tricky, because ultimately we want to change the terminal process's `PATH` _after_ it has finished initializing (e.g. running profile startup scripts). The solution I landed on was to run `bash` as an interactive non-login shell, but then do all the things that a login shell would normally do -- that is, sourcing of `/etc/profile` and `~/.bash_profile`.

Note that the PR, in its current state, only works for the default Bash shell. Will need to confer with @gtritchie if this approach would be workable for other shells as well.

### Automated Tests

None yet.

### QA Notes

To test, try the following:

1. Using the RStudio Global Options dialog, tell RStudio to use a particular copy of Python.
2. Launch a new terminal in RStudio.
3. Run `which python`. Check that it matches the RStudio-configured version of Python.
4. Run `echo $RETICULATE_PYTHON`. Check that it matches the RStudio-configured version of Python.

We should also test that this behavior can be opted-out / disable via the associated option in Tools -> Global Options... -> Terminal.

@gtritchie also note:

> This feature must be tested on both Mac and Windows desktops, and with all available shell types (on Windows they’ll need to install a WSL flavor from Store so it’s available to test). Testing on Linux would be nice, too, but suspect Mac provides sufficient coverage

So this does require a bit of smoke-testing on all of the different flavors of bash we use on different operating systems.

### Checklist
